packages/astro/src/i18n/vite-plugin-i18n.ts

```diff
@@ -10,15 +10,31 @@ type AstroInternalization = {
 	logger: Logger;
 };
 
-export default function astroInternalization({ settings }: AstroInternalization): vite.Plugin {
+export default function astroInternalization({
+	settings,
+	logger,
+}: AstroInternalization): vite.Plugin {
 	return {
 		name: 'astro:i18n',
+		async resolveId(id) {
+			if (id === virtualModuleId) {
+				return resolvedVirtualModuleId;
+			}
+		},
 		load(id) {
 			if (id === resolvedVirtualModuleId) {
 				return `
-					import { getI18nBaseUrl as getI18nBaseUrlInternal } from "astro/i18n";
-								
-					export getI18nBaseUrl = (locale) => getI18nBaseUrlInternal(locale, ${settings.config});
+					import { getI18nBaseUrl as getI18nBaseUrlInternal, getLocalesBaseUrl as _getLocalesBaseUrl } from "astro/i18n";
+					
+					const defaultLocale = ${JSON.stringify(settings.config.experimental.i18n!.defaultLocale)};
+					const locales = ${JSON.stringify(settings.config.experimental.i18n!.locales)};
+					const fallback = ${JSON.stringify(settings.config.experimental.i18n!.fallback)};
+					const base =  ${JSON.stringify(settings.config.base)};
+					const trailingSlash =  ${JSON.stringify(settings.config.trailingSlash)};
+					const format =  ${JSON.stringify(settings.config.build.format)};
```

Comment:
Maybe all those i18n properties could be destructured before the return for readability?

Reply:
I don't understand the comment, can you explain it with some code?

Reply:
You can resolve this one, it wasn't looking great because of the line length in the files tab, my bad!